### PR TITLE
ci: skip re-build again in pretest

### PR DESCRIPTION
**Pull request by virkt25**
current test script run pretest which cleans and rebuilds the project. This isn't needed on CI as
it's a clean build every time.

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,8 @@
     "build:full": "npm run clean:full && npm run bootstrap && npm run build && npm run mocha && npm run lint",
     "pretest": "npm run clean && npm run build:current",
     "test": "node packages/build/bin/run-nyc npm run mocha",
+    "test:ci": "node packages/build/bin/run-nyc npm run mocha",
+    "posttest:ci": "npm run posttest",
```
Comment (raymondfeng) [inline package.json]: Please note `posttest:ci` is not referenced elsewhere. IIUC, the fix should be:

```
"test:ci": "node packages/build/bin/run-nyc npm run mocha && npm run posttest",
```

Comment (raymondfeng) [reply package.json]: `pretest` is not run by `test:ci` in your PR and it fails the build as no compile is done.

**Comment by raymondfeng** [comment]
@virkt25 The `npm run clean` is executed by `lerna` in parallel and it should not bother too much. I'm not sure if it's still worth to have a different script for such little optimization.

**Comment by virkt25** [comment]
@raymondfeng I'm just trying it out (needed to create a PR to trigger CI). Should've left a comment to not review this. We don't have to land this but for travis it seems like it saves about a minute.

File: package.json (hunk 1 above)
Comment (raymondfeng) [reply package.json]: > pretest is not run by test:ci in your PR and it fails the build as no compile is done.

I take it back as the `npm run bootstrap` triggers `prepare` scripts for all packages.

Comment (virkt25) [reply package.json]: Yep. Which in turn triggers a complete build (dist / dist6). The pretest cleans and re-builds the current target again.

**Comment by virkt25** [comment]
@raymondfeng thoughts on landing this PR? I checked times on AppVeyor with previous builds and results are similar to Travis whereby it saves us about a minute on each system.

**Comment by raymondfeng** [comment]
I'm fine to land it.

File: package.json [hunk 2]
```diff
@@ -39,7 +39,8 @@
     "build:full": "npm run clean:full && npm run bootstrap && npm run build && npm run mocha && npm run lint",
     "pretest": "npm run clean && npm run build:current",
     "test": "node packages/build/bin/run-nyc npm run mocha",
-    "mocha": "node packages/build/bin/run-mocha \"packages/*/DIST/test/**/*.js\" \"packages/cli/test/*.js\"",
+    "test:ci": "node packages/build/bin/run-nyc npm run mocha && npm run posttest",
+    "mocha": "node packages/build/bin/select-dist mocha --opts packages/build/mocha.ts.opts \"packages/*/DIST/test/**/*.js\" \"packages/cli/test/*.js\"",
```
Comment (raymondfeng) [inline package.json]: This needs to be reverted.

**Comment by virkt25** [comment]
@raymondfeng fixed. PTAL.

**Pull request merged**